docs/developing.rst

```diff
@@ -596,6 +596,80 @@ Example::
 
             toil.exportFile(outputFileID, 'file:///some/other/local/path')
 
+
+Using Docker containers in Toil
+-------------------------------
+
+Docker containers are commonly used with Toil to maintain the portability of
+workflows. The combination of Toil and Docker allows for pipelines to be fully portable
+between any platform that has both Toil and Docker installed. Docker eliminates the need for
+the user to do any other tool installation or environment setup.
+
+In order to use Docker containers with Toil, Docker must be installed
+on all workers of the cluster in order to run a workflow involving Docker containers.
+Instructions for installing Docker can be found on the `Docker`_ website.
+
+.. _Docker: https://docs.docker.com/engine/getstarted/step_one/
+
+When using Toil-based autoscaling, Docker will be automatically set up on the cluster,
+so no additional installation steps are necessary. Further information on using
+autoscaling can be found in the `Toil autoscaling documentation <Autoscaling>`.
+
+In order to use docker containers in a Toil workflow, the container can be built locally or
+downloaded in real time from an online docker repository like Quay_. If the container is local,
+it must be built on each node of the cluster.
```

Comment:
I think it's fair to say that anyone using a private repository is responsible for figuring that out.